src/instruments/hcp/tc038.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+"""
+Provides support for the TC038 AC crystal oven by HC Photonics.
+"""
+
+
+# IMPORTS #####################################################################
+
+
+from instruments.units import ureg as u
+
+from instruments.abstract_instruments.instrument import Instrument
+from instruments.util_fns import assume_units
+
+import serial
+
+# CLASSES #####################################################################
+
+
+class TC038(Instrument):
+    """
+    Communication with the HCP TC038 oven.
+
+    This is the older version with an AC power supply and AC heater.
+
+    It has parity or framing errors from time to time. Handle them in your
+    application.
+    """
+
+    registers = {
+        "temperature": "D0002",
+        "setpoint": "D0120",
+    }
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the TC038 is a crystal oven.
+
+        Example usage:
+
+        >>> import instruments as ik
+        >>> import instruments.units as u
+        >>> inst = ik.hcp.TC038.open_visa('TCPIP0:192.168.0.35')
+        >>> inst.setpoint = 45.3
+        >>> print(inst.temperature)
+        """
+        super().__init__(*args, **kwargs)
+        self.addr = 1
+
+    @classmethod
+    def open_serial(cls, *args, **kwargs):
+        """Configure the serial connection."""
+        inst = super().open_serial(*args, **kwargs)
+        inst._file._conn.parity = serial.PARITY_EVEN
+        inst._file._terminator = "\r"
+        return inst
```

Comment:
I'm not sure why this is here at all. The `Instrument` class already has a terminator property that could be easily set in the `__init__`. I think the parity should be set in init as well if it is not default. If that is required, it might be better to add a parity property to `instrument` than keeping it here. That's how it is often (usually?) done in other instruments in `ik`.

Reply:
I did not see the terminator property. Thanks for the hint.
Thanks for the suggestion of adding my own property to this instrument. I did not like the implementation, but did not find any better solution.